### ISSUE TITLE
[Outreachy Task Submission] Accessibility Issues in the 'Data View' screen

### ISSUE THREAD

**Issue by Chiemezuo** [ushahidi/platform#4801]
# Intro
Here's a list of some of the Accessibility issues in the 'Data View' screen, in no particular order:
- There's a missing header level on the page. The page has `h1` and proceeds directly to `h3` tags without `h2` tags. WCAG [2.4.6 level AA](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html).
- The search form has a missing label WCAG [2.4.6 level AA](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html).
- All the buttons within the page seem to have no value text or are empty (as flagged by automated software)
- Some individual text items seem to have low contrast.

## Issue
The aforementioned issues will most likely affect users with poor memories or that find it difficult to understand content. Users with screen readers will also encounter difficulty, but I will put that in a separate Github issue, for easier fixing.

## Fix
I'll be working on some fixes to help address some of these issues, but design decisions by the team as a whole will be necessary for problems like low text contrast.
